artifactory/commands/flexpack/maven.go

```diff
@@ -144,63 +180,159 @@ func setMavenBuildPropertiesOnArtifacts(workingDir, buildName, buildNumber strin
 	}
 	defer searchReader.Close()
 
+	// Filter to only artifacts modified in the last 2 minutes (just deployed)
+	cutoffTime := time.Now().Add(-2 * time.Minute)
+	var recentArtifacts []specutils.ResultItem
+
+	for item := new(specutils.ResultItem); searchReader.NextRecord(item) == nil; item = new(specutils.ResultItem) {
+		// Parse the modified time
+		modTime, err := time.Parse("2006-01-02T15:04:05.999Z", item.Modified)
+		if err != nil {
+			log.Debug("Could not parse modified time for " + item.Name + ": " + err.Error())
+			continue
+		}
+
+		// Only include artifacts modified after cutoff
+		if modTime.After(cutoffTime) {
+			recentArtifacts = append(recentArtifacts, *item)
+			log.Debug(fmt.Sprintf("Including recently deployed artifact: %s (modified: %s)", item.Name, item.Modified))
```

Comment:
seeing too many debug logs for build info construction. Would you like to have verbose level?